src/lsdb/catalog/catalog.py

```diff
@@ -68,3 +80,91 @@ def get_partition_index(self, order: int, pixel: int) -> int:
+        matching an inner pixel alignment - using pixels that have overlap in both input catalogs
+        and taking the smallest of any overlapping pixels.
+
+        The resulting catalog will be partitioned using the left catalog's ra and dec, and the
+        index for each row will be the same as the index from the corresponding row in the left
+        catalog's index.
+
+        Args:
+            other (Catalog): The right catalog to cross-match against
+            suffixes (Tuple[str, str]): A pair of suffixes to be appended to the end of each column
+                name when they are joined. Default: uses the name of the catalog for the suffix
+            algorithm (BuiltInCrossmatchAlgorithm | ufunc): The algorithm to use to perform the
+                crossmatch. Can be either a string to specify one of the built-in cross-matching
+                methods, or a custom method defined by subclassing AbstractCrossmatchAlgorithm.
+
+                Built-in methods:
+                    -`kd_tree`: find the k-nearest neighbors using a kd_tree
+
+                Custom function:
+                    To specify a custom function, write a class that subclasses the
+                    `AbstractCrossmatchAlgorithm` class, and overwrite the `crossmatch` function.
+
+                    The function should be able to perform a crossmatch on two pandas DataFrames
+                    from a HEALPix pixel from each catalog. It should return a dataframe with the
+                    combined set of columns from the input dataframes with the appropriate suffixes,
+                    and a column `_DIST` with the distance between the points.
+
+                    The class will have been initialized with the following parameters, which the
+                    crossmatch function should use:
+
+                        left: pd.DataFrame,
+                        right: pd.DataFrame,
+                        left_order: int,
+                        left_pixel: int,
+                        right_order: int,
+                        right_pixel: int,
+                        left_metadata: hc.catalog.Catalog,
+                        right_metadata: hc.catalog.Catalog,
+                        suffixes: Tuple[str, str]
+
+                    You may add any additional keyword argument parameters to the crossmatch
+                    function definition, and the user will be able to pass them in as kwargs in the
+                    `Catalog.crossmatch` method.
+
+            name (str): The name of the resulting catalog. Default: {left_name}_x_{right_name}
+
+        Returns:
+            A Catalog with the data from the left and right catalogs merged with one row for each
+            pair of neighbors found from cross-matching.
+
+            The resulting table contains all columns from the left and right catalogs with their
+            respective suffixes, and a `_DIST` column with the great circle separation between the
```

Comment:
Maybe make `"_DIST"` a const on the abstract base class to encourage consistent use?

Reply:
Wooo a string constant comment. Never been so happy to make a PR change